modules/programs/wpaperd.nix

```
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.wpaperd;
+  tomlFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ maintainers.Avimitin ];
+
+  options.programs.wpaperd = {
+    enable = mkEnableOption "wpaperd";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.wpaperd;
+      defaultText = literalExpression "pkgs.wpaperd";
+      description = "wpaperd package to install.";
+    };
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
+      example = {
+        eDP-1 = {
+          path = "/home/foo/Pictures/Wallpaper";
+          apply-shadow = true;
+        };
+        DP-2 = {
+          path = "/home/foo/Pictures/Anime";
+          sorting = "descending";
+        };
+      };
```

Best to make this a literal expression since it's a pretty long expression (right now it will be rendered as a single line).

```suggestion
      example = literalExpression ''
        {
          eDP-1 = {
            path = "/home/foo/Pictures/Wallpaper";
            apply-shadow = true;
          };
          DP-2 = {
            path = "/home/foo/Pictures/Anime";
            sorting = "descending";
          };
        };
      '';
```